File: app/src/main/java/com/muugi/shortcut/sample/WelcomeActivity.java
package com.muugi.shortcut.sample;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.muugi.shortcut.core.Shortcut;

import com.muugi.shortcut.setting.ShortcutPermission;

/**
 * Created by ZP on 2019-06-16.
 */
public class WelcomeActivity extends AppCompatActivity {

    private TextView mTvShortcutPermissionState;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);
        App.isLogined = true;
        initViews();
    }

    private void initViews() {
        mTvShortcutPermissionState = findViewById(R.id.tv_shortcut_permission_state);
        mTvShortcutPermissionState.setOnClickListener(v -> {
            Shortcut.Companion.getSingleInstance().openSetting(WelcomeActivity.this);
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        int check = ShortcutPermission.check(this);
        String state = "未知";
        switch (check) {
            case ShortcutPermission.PERMISSION_DENIED:
                state = "已禁止";
                break;
            case ShortcutPermission.PERMISSION_GRANTED:
                state = "已同意";
                break;
            case ShortcutPermission.PERMISSION_ASK:
                state = "询问";
                break;
            case ShortcutPermission.PERMISSION_UNKNOWN:
                state = "未知";
                break;
        }
        mTvShortcutPermissionState.setText(state);
    }

    public void goToCreateShortcut(View view) {
        startActivity(new Intent(this, MainActivityV2.class));
    }

    public void goToPreview(View view) {
        Intent intent = new Intent(this, PreviewActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/muugi/shortcut/sample/WelcomeActivity.kt
package com.muugi.shortcut.sample

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.Button
import androidx.compose.material.ButtonDefaults
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.muugi.shortcut.core.Shortcut
import com.muugi.shortcut.setting.ShortcutPermission

/**
 * Created by ZP on 2019-06-16.
 */
class WelcomeActivity : ComponentActivity() {

    val shortcutViewModel by lazy {
        ShortcutViewModel()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            Column {
                val collectAsState = shortcutViewModel.permissionState.collectAsState()
                ItemAction("创建桌面快捷方式权限", collectAsState.value) {
                    Shortcut.singleInstance.openSetting(this@WelcomeActivity)
                }

                ItemAction("预览桌面快捷方式", "预览") {
                    goToPreview()
                }
                Button(
                    colors = ButtonDefaults.buttonColors(backgroundColor = Color(0xFF00574B)),
                    modifier = Modifier.fillMaxWidth().height(56.dp).padding(10.dp), onClick = {
                        goToCreateShortcut()
                    }) {
                    Text(text = "创建快捷方式", color = Color.White, fontSize = 16.sp)
                }
            }
        }
        App.isLogined = true
    }

    @Composable
    fun ItemAction(actionName: String, actionState: String, onClick: () -> Unit) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier.padding(horizontal = 16.dp).fillMaxWidth().height(68.dp)
                .clickable { onClick() }) {
            Text(text = actionName, color = Color(0xFF2a2a2a), fontSize = 18.sp)
            Spacer(Modifier.fillMaxWidth().weight(1.0f))
            Text(text = actionState, color = Color(0xFFD5D5D5), fontSize = 15.sp)
        }
    }


    override fun onResume() {
        super.onResume()
        shortcutViewModel.checkPermission(this)
    }

    fun goToCreateShortcut() {
        startActivity(Intent(this, MainActivity::class.java))
    }

    fun goToPreview() {
        val intent = Intent(this, PreviewActivity::class.java)
        startActivity(intent)
    }
}